brainframe_qt/ui/brainframe_app.py

```diff
@@ -1,98 +1,92 @@
-        translator = QTranslator(self)
-        i18n_dir = str(text_paths.i18n_dir)
-        if not translator.load(locale, "brainframe", '_', i18n_dir):
-
-            if locale.language() != locale.English:
-                # TODO: Find a better way that doesn't rely on a _list of
-                #  preferences_. If, say, zh_BA (which isn't a thing) is
-                #  used, locale.name() returns zh_CN (which _is_ a real
-                #  language). If there is no `zh` language, we want to throw
-                #  a warning with zh_BA specified, not zh_CN. We settle for
-                #  locale.uiLanguages() because it returns ['zh_BA']. Not
-                #  sure if it might have other entries under some conditions.
-                locale_str = locale.uiLanguages()[0]
-
-                title = "Error loading language files"
-                message = (f"Unable to load translation file for "
-                           f"[{locale_str}] locale. Using English as a "
-                           f"fallback.")
-
-                # noinspection PyTypeChecker
-                BrainFrameMessage.warning(
-                    parent=None,
-                    title=title,
-                    warning=message
-                ).exec()
+    def _init_signals(self) -> None:
+        self.aboutToQuit.connect(self._shutdown)
+        self.connection_manager.connection_state_changed.connect(
+            self._on_connection_state_change
+        )
 
-        else:
-            self.installTranslator(translator)
+        # If the splashscreen is manually closed, we should just exit
+        self.splash_screen.manually_closed.connect(self.quit)
 
-    def exec(self):
+    def _init_style(self) -> None:
+        self.setWindowIcon(QIcon(":/icons/window_icon"))
 
-        self._verify_eula()
+    def _init_translator(self) -> BrainFrameTranslator:
+        translator = BrainFrameTranslator()
+        self.installTranslator(translator)
 
-        # Show splash screen while waiting for server connection
-        with SplashScreen() as self.splash_screen:
-            self._connect_to_server()
-            self._verify_version_match()
+        return translator
 
-            message = self.tr("Successfully connected to server. Starting UI")
-            self.splash_screen.showMessage(message)
+    def exec(self):
+        self._verify_eula()
+        self.connection_manager.start()
+        self.splash_screen.show()
+        super().exec()
 
-            ExtensionLoader().load_extensions()
+    def _on_connection_state_change(
+        self, connection_state: ConnectionManager.ConnectionState
+    ):
+        if connection_state is ConnectionManager.ConnectionState.CONNECTED:
+            message = self.tr("Successfully connected to server")
+            self._start_ui()
+        elif connection_state is ConnectionManager.ConnectionState.UNCONFIGURED:
+            message = self.tr("Collecting server authentication configuration")
```

Comment:
Out of curiosity, do users actually ever see this message in practice? I assume the operation is pretty darn fast but I want to make sure I understand what this state means. Regardless, I still think we should keep it.

Reply:
In actuality, it's gonna only appear for a single frame, at most

Reply:
Cool